float32[] ranges
int32 query_id
